openpype/hosts/maya/tools/mayalookassigner/arnold_standin.py

```diff
@@ -0,0 +1,279 @@
+            ))
+            continue
+
+        relationships = lib.get_look_relationships(version["_id"])
+        shader_nodes, container_node = lib.load_look(version["_id"])
+        namespace = shader_nodes[0].split(":")[0]
+
+        # Get only the node ids and paths related to this asset
+        # And get the shader edits the look supplies
+        asset_nodes_by_id = {
+            node_id: nodes_by_id[node_id] for node_id in node_ids
+        }
+        edits = list(
+            api.lib.iter_shader_edits(
+                relationships, shader_nodes, asset_nodes_by_id
+            )
+        )
+
+        # Create assignments
+        node_assignments = {}
+        for edit in edits:
+            for node in edit["nodes"]:
+                if node not in node_assignments:
+                    node_assignments[node] = []
+
+            if edit["action"] == "assign":
+                if not cmds.ls(edit["shader"], type="shadingEngine"):
+                    log.info("Skipping non-shader: %s" % edit["shader"])
+                    continue
+
+                shading_engine_assignments(
+                    edit["shader"],
+                    "surfaceShader",
+                    edit["nodes"],
+                    node_assignments
+                )
+                shading_engine_assignments(
+                    edit["shader"],
+                    "displacementShader",
+                    edit["nodes"],
+                    node_assignments
+                )
+
+            if edit["action"] == "setattr":
+                visibility = False
+                for attr, value in edit["attributes"].items():
+                    if attr not in ATTRIBUTE_MAPPING:
+                        log.warning(
+                            "Skipping setting attribute {} on {} because it is"
+                            " not recognized.".format(attr, edit["nodes"])
+                        )
+                        continue
+
+                    if isinstance(value, str):
+                        value = "'{}'".format(value)
+
+                    if ATTRIBUTE_MAPPING[attr] == "visibility":
+                        visibility = True
+                        continue
+
+                    assignment = "{}={}".format(ATTRIBUTE_MAPPING[attr], value)
+
+                    for node in edit["nodes"]:
+                        node_assignments[node].append(assignment)
+
+                if visibility:
+                    # https://arnoldsupport.com/2018/11/21/backdoor-setting-visibility/
+                    mapping = {
+                        "primaryVisibility": 1,  # Camera
+                        "castsShadows": 2,  # Shadow
+                        "aiVisibleInDiffuseTransmission": 4,
+                        "aiVisibleInSpecularTransmission": 8,
+                        "aiVisibleInVolume": 16,
+                        "aiVisibleInDiffuseReflection": 32,
+                        "aiVisibleInSpecularReflection": 64
+                    }
```

Comment:
Maybe best to make this global to the file to use it here and above in the function instead of duplicating it?

**Edit:** Or actually, should this just be: `calculate_visibility_mask(edit["attributes"])`?

Reply:
Lol, yeah. Must have forgotten to complete the refactor.